node/core/candidate-validation/src/lib.rs

```diff
@@ -321,52 +322,52 @@ async fn spawn_validate_exhaustive(
 	rx.await.map_err(Into::into)
 }
 
-/// Does basic checks of a candidate. Provide the encoded PoV-block. Returns `true` if basic checks
-/// are passed, false otherwise.
-fn passes_basic_checks(
+/// Does basic checks of a candidate. Provide the encoded PoV-block. Returns `Ok` if basic checks
+/// are passed, `Err` otherwise.
+fn perform_basic_checks(
 	candidate: &CandidateDescriptor,
 	max_block_data_size: Option<u64>,
 	pov: &PoV,
-) -> bool {
+) -> Result<(), InvalidCandidate> {
 	let encoded_pov = pov.encode();
 	let hash = pov.hash();
 
 	if let Some(max_size) = max_block_data_size {
 		if encoded_pov.len() as u64 > max_size {
-			return false;
+			return Err(InvalidCandidate::ParamsTooLarge(encoded_pov.len() as u64));
```

Comment:
This `max_block_data_size` is not a protocol limit, but we could make it one so that there is consensus on it.